appinventor/components/src/com/google/appinventor/components/runtime/Clock.java

```diff
@@ -207,6 +216,92 @@ public static Calendar MakeInstant(String from) {
     }
   }
 
+  /**
+   * An instant in time specified by number year, number month, number day
+   * @param year year integer
+   * @param month month integer
+   * @param day day integer
+   * @return  Calendar instant
+   */
+  @SimpleFunction(description = "Allows the user to set the clock to be "
+    +"a date value.\n" +
+    "Valid values for the month field are 1-12 and 1-31 for the day field.\n")
+  public Calendar MakeDate(int year, int month, int day) {
+    int jMonth = month - 1;
+    try {
+      GregorianCalendar cal = new GregorianCalendar(year, jMonth, day);
+      cal.setLenient(false);
+      cal.getTime();
+    } catch (java.lang.IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, "MakeDate", ErrorMessages.ERROR_ILLEGAL_DATE);
+    }
+    
+    Calendar instant = Dates.DateInstant(year, month, day);
+    return instant;
+  }
+
+  /**
+   * An instant in time specified by integer hour and integer minute
+   * @param hour hour integer
+   * @param minute minute integer
+   * @return  Calendar instant since 1/1/1970
+   */
+  @SimpleFunction(description = "Allows the user to set the time of the clock - " +
+    "Valid format is hh:mm\n")
+  public Calendar MakeTime(int hour, int minute) {
+    Calendar instant = new GregorianCalendar();
+    try {
+      instant.set(Calendar.HOUR_OF_DAY, hour);
+      instant.set(Calendar.MINUTE, minute);
+      instant.set(Calendar.SECOND, 0);
+    } catch (java.lang.IllegalArgumentException e) {
```

Comment:
You shouldn't need to fully qualify IllegalArgumentException here because java.lang is always available.